sdk/core/core-rest-pipeline/src/policies/proxyPolicy.ts

```diff
@@ -15,6 +17,10 @@ import {
 import { PipelinePolicy } from "../pipeline";
 import { URL } from "../util/url";
 
+// During rush build:test, these two appear not to be exported by the module.
+const { HttpsProxyAgent } = httpsAgent;
+const { HttpProxyAgent } = httpAgent;
+
```

Comment:
@xirzec Hello hello! Without this change, the tests for packages like `keyvault-secrets` wouldn't build in this PR 🤔 the error I get says that `HttpsProxyAgent` and `HttpProxyAgent` are not exported by their respective packages.

Reply:
Are the tests for keyvault-secrets failing even in the main branch or only after moving Identity to core-rest-pipeline. If it is the former, can we have a separate PR to fix that problem and not mix it into the work where we move Identity to core-rest-pipeline?

Reply:
Tests fail to build (on keyvault-secrets and like 3 other packages) only after moving Identity to core-rest-pipeline

Reply:
Can we please investigate why the tests fail to build? I understand that you see the error `HttpsProxyAgent and HttpProxyAgent are not exported`, but why do the tests even need them? We have shipped GA packages that depend on core-rest-pipeline without this change. What is different about the tests that they need this change?

If the investigation needs more time and we do not have time in this milestone, am ok moving the task of migrating Identity to core-rest-pipeline to the next milestone. That would be preferable than making changes that we do not fully understand the need for.

Reply:
These changes seem very strange to me, certainly I wouldn't expect them to be necessary.

Reply:
This doesn't make any sense.

HttpsProxyAgent isn't even exported by this module, so how would it affect the package's surface to change the form of the import? I suspect something else (leftover compilation state?) is at play here and these changes are just a red herring.

Reply:
Let’s study this. I’ve seen this outside of the package at least once, but I can’t remember the details. I’ll ping @witemple-msft today.

Reply:
After resetting my environment and installing everything again in a fresh state, the problem went away!

Reply:
I’ve removed this change.

Reply:
Oh! News. After I removed the changes, the bug reproduced on CI: https://dev.azure.com/azure-sdk/public/_build/results?buildId=1021300&view=logs&j=3cacf272-7c7b-57a7-ca0c-8c0bd3bac579&t=7fd19a33-2f2c-535e-e5ae-1da244c27272 (I can’t reproduce this locally)

Reply:
Turns out I needed to add `"http-proxy-agent", "https-proxy-agent"` on the `external` dependencies on the rollup files for some packages, including keyvault-secrets. Doing that!

These packages need to move to the shared rollup. The shared rollup doesn’t have this problem (but doing that migration is not straightforward).

Reply:
Adding `@azure/identity` to `external` also works!